addons/website_sale/static/src/js/website_sale.js

```diff
@@ -193,6 +193,7 @@ $('.oe_website_sale').each(function () {
             $product_id.val(0);
             $parent.find(".js_check_product").attr("disabled", "disabled");
         }
+        $product_id.trigger('change');
```

Comment:
I guess this will trigger the js code of `website_sale_stock` ? 
Maybe you should make a odoo widget instead of native js code, and extend it in website_sale_stock.
Do it in a separate commit.